presto-native-execution/presto_cpp/main/connectors/tpcds/TpcdsGen.cpp

```diff
@@ -0,0 +1,945 @@
+     BIGINT(),      BIGINT(),      BIGINT(),      INTEGER(),     DECIMAL(7, 2),
+     DECIMAL(7, 2), DECIMAL(7, 2), DECIMAL(7, 2), DECIMAL(7, 2), DECIMAL(7, 2),
+     DECIMAL(7, 2), DECIMAL(7, 2), DECIMAL(7, 2), DECIMAL(7, 2), DECIMAL(7, 2),
+     DECIMAL(7, 2), DECIMAL(7, 2), DECIMAL(7, 2), DECIMAL(7, 2)});
+
+static const RowTypePtr websiteType = ROW(
+    {"web_site_sk",        "web_site_id",       "web_rec_start_date",
+     "web_rec_end_date",   "web_name",          "web_open_date_sk",
+     "web_close_date_sk",  "web_class",         "web_manager",
+     "web_mkt_id",         "web_mkt_class",     "web_mkt_desc",
+     "web_market_manager", "web_company_id",    "web_company_name",
+     "web_street_number",  "web_street_name",   "web_street_type",
+     "web_suite_number",   "web_city",          "web_county",
+     "web_state",          "web_zip",           "web_country",
+     "web_gmt_offset",     "web_tax_percentage"},
+    {BIGINT(),      VARCHAR(),    DATE(),    DATE(),    VARCHAR(), BIGINT(),
+     BIGINT(),      VARCHAR(),    VARCHAR(), INTEGER(), VARCHAR(), VARCHAR(),
+     VARCHAR(),     INTEGER(),    VARCHAR(), VARCHAR(), VARCHAR(), VARCHAR(),
+     VARCHAR(),     VARCHAR(),    VARCHAR(), VARCHAR(), VARCHAR(), VARCHAR(),
+     DECIMAL(5, 2), DECIMAL(5, 2)});
+
+const RowTypePtr getTableSchema(Table table) {
+  switch (table) {
+    case Table::TBL_CALL_CENTER:
+      return callCenterType;
+    case Table::TBL_CATALOG_PAGE:
+      return catalogPageType;
+    case Table::TBL_CATALOG_RETURNS:
+      return catalogReturnsType;
+    case Table::TBL_CATALOG_SALES:
+      return catalogSalesType;
+    case Table::TBL_CUSTOMER:
+      return customerType;
+    case Table::TBL_CUSTOMER_ADDRESS:
+      return customerAddressType;
+    case Table::TBL_CUSTOMER_DEMOGRAPHICS:
+      return customerDemographicsType;
+    case Table::TBL_DATE_DIM:
+      return dateDimType;
+    case Table::TBL_HOUSEHOLD_DEMOGRAPHICS:
+      return householdDemographicsType;
+    case Table::TBL_INCOME_BAND:
+      return incomeBandType;
+    case Table::TBL_INVENTORY:
+      return inventoryType;
+    case Table::TBL_ITEM:
+      return itemType;
+    case Table::TBL_PROMOTION:
+      return promotionType;
+    case Table::TBL_REASON:
+      return reasonType;
+    case Table::TBL_SHIP_MODE:
+      return shipModeType;
+    case Table::TBL_STORE:
+      return storeType;
+    case Table::TBL_STORE_RETURNS:
+      return storeReturnsType;
+    case Table::TBL_STORE_SALES:
+      return storeSalesType;
+    case Table::TBL_TIME_DIM:
+      return timeDimType;
+    case Table::TBL_WAREHOUSE:
+      return warehouseType;
+    case Table::TBL_WEB_PAGE:
+      return webPageType;
+    case Table::TBL_WEB_RETURNS:
+      return webReturnsType;
+    case Table::TBL_WEB_SALES:
+      return webSalesType;
+    case Table::TBL_WEB_SITE:
+      return websiteType;
+    default:
+      VELOX_UNREACHABLE();
+  }
+}
+
+std::string toTableName(Table table) {
```

Comment:
Use the reverse of the map in fromTableName for toTableName. PlanNode.cpp has some good examples of the usage of such maps e.g. https://github.com/facebookincubator/velox/blob/main/velox/core/PlanNode.cpp#L1452